src/NServiceBus.Core/Encryption/RijndaelEncryptionService.cs

```diff
@@ -227,6 +227,11 @@ protected virtual bool TryGetKeyIdentifierHeader(out string keyIdentifier, IInco
             return context.Headers.TryGetValue(Headers.RijndaelKeyIdentifier, out keyIdentifier);
         }
 
+        protected virtual void GenerateIV(RijndaelManaged rijndael)
```

Comment:
I don't really like the fact that the Rijndael instance is exposed when the class is inherited but as it is private I don't really see an issue with it in this context.

Just to be complete, an alternative could be:
```c#
static RNGCryptoServiceProvider random = new RNGCryptoServiceProvider(); // Thread-safe

protected byte[] GenerateIV(int length)
{
    var iv = new byte[length];
    random.GetBytes(iv);
    return iv;
}
```

That required to understand more about the Security namespace. I like @MarcinHoppe his approach for effectiveness and readability.